File: Python/DS/stack.py
class Stack:
    """
    Stack class represents a stack data structure.
    It contains methods for pushing, popping, checking if it's empty,
    peeking at the top element, and getting the size of the stack.
    """
    def __init__(self):
        self.items = []

    def push(self, item):
        """
        Push an item onto the stack.
        """
        self.items.append(item)

    def pop(self):
        """
        Pop and return the top item from the stack.
        """
        if not self.is_empty():
            return self.items.pop()
        else:
            return None

    def is_empty(self):
        """
        Check if the stack is empty.
        """
        return len(self.items) == 0

    def peek(self):
        """
        Return the top item from the stack without removing it.
        """
        if not self.is_empty():
            return self.items[-1]
        else:
            return None

    def size(self):
        """
        Return the size (number of items) of the stack.
        """
        return len(self.items)

# Example usage:
if __name__ == "__main__":
    stack = Stack()
    stack.push(1)
    stack.push(2)
    stack.push(3)
    print("Stack:", stack.items)
    print("Peek:", stack.peek())
    print("Pop:", stack.pop())
    print("Stack Size:", stack.size())

File: JavaScript/DS/stack.js
/**
 * Class representing a Stack data structure.
 */
class Stack {
    /**
     * Create an empty stack.
     */
    constructor() {
      this.items = [];
    }
  
    /**
     * Push an item onto the stack.
     * @param {any} item - The item to be pushed onto the stack.
     */
    push(item) {
      this.items.push(item);
    }
  
    /**
     * Pop and return the top item from the stack.
     * @returns {any} - The top item from the stack.
     */
    pop() {
      if (!this.isEmpty()) {
        return this.items.pop();
      } else {
        return null;
      }
    }
  
    /**
     * Check if the stack is empty.
     * @returns {boolean} - True if the stack is empty, otherwise false.
     */
    isEmpty() {
      return this.items.length === 0;
    }
  
    /**
     * Return the top item from the stack without removing it.
     * @returns {any} - The top item from the stack.
     */
    peek() {
      if (!this.isEmpty()) {
        return this.items[this.items.length - 1];
      } else {
        return null;
      }
    }
  
    /**
     * Return the size (number of items) of the stack.
     * @returns {number} - The size of the stack.
     */
    size() {
      return this.items.length;
    }
  }
  
  module.exports = Stack;
  
  // Example usage:
  if (require.main === module) {
    const stack = new Stack();
    stack.push(1);
    stack.push(2);
    stack.push(3);
    console.log("Stack:", stack.items);
    console.log("Peek:", stack.peek());
    console.log("Pop:", stack.pop());
    console.log("Stack Size:", stack.size());
  }
  